### DISCUSSION TITLE
Help regarding the fast-downward planner

### DISCUSSION THREAD

**Discussion by Drejer1** [aiplan4eu/unified-planning#519] [Q&A] [answered]
So I have tried following the getting started guide, to solve problems. I have some solvable problems that works fine whenever I use the pyperplan engine but when I switch to fast-downward the problems becomes unsolvable for some reason. I have tried running the fast-downward planner outside of the unified-planning library on the problems and it works fine then.  I'm pretty sure the engine is installed correctly, at least the credits show up when running the planner. 
Looking at the docs https://unified-planning.readthedocs.io/en/latest/engines/3-fast_downward.html I'm not seeing anything I'm missing at least.
```python
def check_if_solvable(problem: Problem):
    with OneshotPlanner(name='fast-downward') as planner:
        result = planner.solve(problem)
        if result.status == up.engines.PlanGenerationResultStatus.SOLVED_SATISFICING:
            # print("Pyperplan returned: %s" % result1.plan)
            return True
        else:
            print("No plan found.")
            return False
```

**Comment by Framba-Luca** [comment]
Hi @Drejer1 , can you provide some (or all) the problem examples? It really helps debugging :smiley: 

In general, add all the code you can. Thank you very much!

**Reply by Drejer1** [reply]
It fails already on the first one, problem so here they are. 

[domain_best.txt](https://github.com/aiplan4eu/unified-planning/files/13335746/domain_best.txt)
[p01.txt](https://github.com/aiplan4eu/unified-planning/files/13335744/p01.txt)

**Reply by Framba-Luca** [reply]
Thanks, having a look at it!

**Reply by Drejer1** [reply]
I'm the one thanking you for your time.

**Reply by Framba-Luca** [reply]
To me it correctly works both on the version I had installed (the latest one, 0.3.1 for up-fast-downward and 1.0.0.166.post1 for UP) and on the version compatible with up 1.0.0 (0.2.3), so you probably have some issue with the installation.

Try to delete everything and then re-install up and fast-downward. If it still does not work list here some info about your machine (like OS and stuff like that).

I installed it with this command: `pip3 install unified-planning[fast-downward]==1.0.0`

**Reply by Drejer1** [reply]
I have now tried reinstalling Python, Pycharm, and installed the packages again. Running it in both virtual and local environment.
Still it does not work. The parameter arbimo gave me spits this message out when running it.
´´´
Could not find build 'release' at C:\Users\Drejer\PycharmProjects\PDDL_library_Tests\venv\lib\site-packages\up_fast_downward\downward\builds\release\bin. Please run './build.py release'.
´´´
I'm running windows 10 on a 64-bit machine. Is there any other relevant information?

**Reply by Framba-Luca** [reply]
Try finding the script `build.py` and run `./build.py release`, as suggested in the error message.

**Reply by Drejer1** [reply]
I have multiple files called build.py, non of them atleast seems to be related to the fast-downward lib. There is not a script related to up-fast-downward in ```venv/scripts``` at all. 
Running ``` ./build.py release``` in the project paths which do contain a build.py 
``` venv/Lib/site-packages/setuptools/command```
``` venv/Lib/site-packages/setuptools/_distutils/command```
just gives me ``` ImportError: attempted relative import with no known parent package```.

I have tried following the manual install on https://unified-planning.readthedocs.io/en/latest/engines/3-fast_downward.html skipping the first step since currently that link brings me to a dead page (which is weird since I'm pretty sure I looked at it a month or 2 ago) , but fast-downward works when I'm running that in terminal alone, so I assume the dependencies function. Doing it that way started giving me assertion errors when starting the program, so I discarded that idea. 

I just noticed that when I'm installing pyperplan I get both a package named "pyperplan" and "up-pyperplan" while installing fast-downward I only get "up-fast-downward" I don't know if that means anything? 

I will very much admit I am a novice at python, so I gotta ask. There is not anything from this[ page ](https://unified-planning.readthedocs.io/en/latest/engines/3-fast_downward.html )I'm missing is there? Something so obvious I am completely missing it? It really should just be writing ```pip install unified-planning[fast-downward]```?  So why am I missing the build.py related to fast-downward?

**Reply by Framba-Luca** [reply]
Is fast-downward the only engine you are having problems with? 

@roeger Maybe you have seen this problem already?

**Reply by roeger** [reply, marked as answer]
Thanks for the heads up @Framba-Luca, I didn't see this before. This issue might be related to a bug in building the wheels for windows and was fixed in version 0.3.1 of up-fast-downward. @Drejer1, can you please install that version and try it?
`pip install up-fast-downward==0.3.1`

**Reply by Drejer1** [reply]
It works now, with the use of the 0.3.1 version. 
Thank you for your help @Framba-Luca  and @roeger

**Comment by arbimo** [comment]
Could you give the full print of `result` ? Note that there are several positive outcomes (e.g. `SOLVED_OPTIMALLY`) that may be associated with a plan. For a negative result, it would be also interesting to see the log of the planner probably (parameter `output_stream=sys.stdout` to the `solve` method).

**Reply by Drejer1** [reply]
Is this what you mean when you say full print of result? 

```
status: INTERNAL_ERROR
engine: Fast Downward
plan: None
```


Using the parameter you gave gives this output.
I assume this might be the error here.
```
Usage: python implementation.py <path>
Could not find build 'release' at C:\Users\Drejer\PycharmProjects\PDDL_library_Tests\venv\lib\site-packages\up_fast_downward\downward\builds\release\bin. Please run './build.py release'.
INFO     planner time limit: None
INFO     planner memory limit: None

INFO     Running translator.
```
